contracts/flow/token/BloctoToken.cdc

```diff
@@ -70,115 +153,172 @@ pub contract BloctoToken: FungibleToken {
         // It is allowed to destroy the sent Vault because the Vault
         // was a temporary holder of the tokens. The Vault's balance has
         // been consumed and therefore can be destroyed.
-        pub fun deposit(from: @FungibleToken.Vault) {
+        access(all)
+        fun deposit(from: @{FungibleToken.Vault}): Void{ 
             let vault <- from as! @BloctoToken.Vault
             self.balance = self.balance + vault.balance
             emit TokensDeposited(amount: vault.balance, to: self.owner?.address)
             vault.balance = 0.0
             destroy vault
         }
+        
+        access(all)
+        fun createEmptyVault(): @{FungibleToken.Vault}{ 
+            return <-create Vault(balance: 0.0)
+        }
+        
+        access(all)
+        view fun isAvailableToWithdraw(amount: UFix64): Bool{ 
+            return self.balance >= amount
+        }
+
+
+        // Called when a fungible token is burned via the `Burner.burn()` method
+        access(contract) fun burnCallback() {
+            if self.balance > 0.0 {
+                BloctoToken.totalSupply = BloctoToken.totalSupply - self.balance
+            }
+            self.balance = 0.0
+        }
 
-        destroy() {
-            BloctoToken.totalSupply = BloctoToken.totalSupply - self.balance
+        // getSupportedVaultTypes optionally returns a list of vault types that this receiver accepts
+        access(all) view fun getSupportedVaultTypes(): {Type: Bool} {
+            return {self.getType(): true}
         }
-    }
 
+        access(all) view fun isSupportedVaultType(type: Type): Bool {
+            if (type == self.getType()) { return true } else { return false }
+        }
+
+
+         // Get all the Metadata Views implemented by FlowToken
+        //
+        // @return An array of Types defining the implemented views. This value will be used by
+        //         developers to know which parameter to pass to the resolveView() method.
+        //
+        access(all) view fun getViews(): [Type]{
+            return BloctoToken.getContractViews(resourceType: nil)
+        }
+
+        // Get a Metadata View from FlowToken
+        //
+        // @param view: The Type of the desired view.
+        // @return A structure representing the requested view.
+        //
+        access(all) fun resolveView(_ view: Type): AnyStruct? {
+            return BloctoToken.resolveContractView(resourceType: nil, viewType: view)
+        }
+    }
+    
     // createEmptyVault
     //
     // Function that creates a new Vault with a balance of zero
     // and returns it to the calling context. A user must call this function
     // and store the returned Vault in their storage in order to allow their
     // account to be able to receive deposits of this token type.
     //
-    pub fun createEmptyVault(): @FungibleToken.Vault {
+    access(all)
+    fun createEmptyVault(vaultType: Type): @{FungibleToken.Vault}{ 
```

Comment:
`fun createEmptyVault(vaultType: Type):` should return own type `@BloctoToken.Vault` instead of generic type `@{FungibleToken.Vault}`

Reference:
https://github.com/onflow/flow-core-contracts/blob/stable-cadence/contracts/FlowToken.cdc#L165-L167